test/object-store/sync/flx_sync.cpp

```diff
@@ -4202,6 +4202,165 @@ TEST_CASE("flx: compensating write errors get re-sent across sessions", "[sync][
     REQUIRE(top_level_table->is_empty());
 }
 
+TEST_CASE("flx: compensating write errors are not duplicated", "[sync][flx][compensating write][baas]") {
+    FLXSyncTestHarness harness("flx_compensating_writes");
+    auto config = harness.make_test_file();
+
+    auto test_obj_id_1 = ObjectId::gen();
+    auto test_obj_id_2 = ObjectId::gen();
+
+    enum class TestState { Start, FirstError, SecondError, Resume, ThirdError, FourthError };
+    TestingStateMachine<TestState> state(TestState::Start);
+
+    std::mutex errors_mutex;
+    std::vector<std::pair<ObjectId, sync::version_type>> error_to_download_version;
+    std::vector<sync::CompensatingWriteErrorInfo> compensating_writes;
+    sync::version_type download_version;
+
+    config.sync_config->on_sync_client_event_hook = [&](std::weak_ptr<SyncSession> weak_session,
+                                                        const SyncClientHookData& data) {
+        if (auto session = weak_session.lock(); !session) {
+            return SyncClientHookAction::NoAction;
```

Comment:
looking at the coverage report for this test, it looks like this never happens. can we just assert that the session is okay? or just not check it at all?

Reply:
this check, as in other places, is for when the realm and session instances are torn down by the test harness. sometimes it happens that the hook is invoked and captured variables are used out of scope.